src/client/pages/Geo/GeoMap/GeoMapMenuData/MapVisualizerPanel/LayersSectionPanel/components/CustomAssetControl/CustomAssetControl.tsx

```diff
@@ -52,6 +55,13 @@ const CustomAssetControl: React.FC<Props> = ({
     }
   }
 
+  const checkboxStyle = checked && loadingStatus !== LayerFetchStatus.Loading ? { backgroundColor } : {}
+  let checkBoxClass = classNames('fra-checkbox', { checked })
+  if (loadingStatus === LayerFetchStatus.Loading) {
+    checkBoxClass = 'loading-spinner'
+  } else if (loadingStatus === LayerFetchStatus.Failed) {
+    checkBoxClass = classNames('fra-checkbox', 'failed')
+  }
```

Comment:
this can be defined directly within classNames attribute and it can be:
```
classNames({ 'fra-checkbox': loadingStatus !== LayerFetchStatus.Loading, checked, 'loading-spinner': loadingStatus === LayerFetchStatus.Loading, 'failed': loadingStatus === LayerFetchStatus.Failed })
```
have a look here: https://github.com/JedWatson/classnames

Reply:
Way better, thanks!